subsys/net/lib/dns/mdns_responder.c

```diff
@@ -684,13 +750,82 @@ static int init_listener(void)
 	return !ok;
 }
 
-static int mdns_responder_init(void)
+static void mdns_net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+					uint32_t mgmt_event, struct net_if *iface)
+
 {
+	int ret;
+
+	ARG_UNUSED(ret); /* Potentially not used */
 
-	net_mgmt_init_event_callback(&mgmt_cb, mdns_iface_event_handler,
+	switch (mgmt_event) {
+	case NET_EVENT_IF_UP:
+
+#if defined(CONFIG_NET_IPV4)
```

Comment:
So I propose that all the processing in this function is moved to separate k_work so that we avoid weird errors if the mgmt thread stack size is too small.

Reply:
Thanks for the comments, I was looking into refactoring with a `k_work_delayable` in a future PR, to start taking timing requirements into account.
I'll try to simplify it here without the timings.